frontend/src/components/classify-failures.js

```diff
@@ -80,184 +151,173 @@ export class ClassifyFailuresTable extends React.Component {
         hideTestObject=false;
         defaultTab='summary';
       }
-      rows[rowIndex + 1].cells = [{
-        title: <ResultView defaultTab={defaultTab} hideTestHistory={false} hideSummary={hideSummary} hideTestObject={hideTestObject} testResult={rows[rowIndex].result}/>
-      }];
-    }
-    rows[rowIndex].isOpen = isOpen;
-    this.setState({rows});
-  }
 
-  onTableRowSelect = (event, isSelected, rowId) => {
-    let rows;
-    if (rowId === -1) {
-      rows = this.state.rows.map(oneRow => {
-        oneRow.selected = isSelected;
-        return oneRow;
+      const updatedRows = rows.map((row, index) => {
+        let newRow = {};
+        // set isOpen on the parent row items
+        if (index === rowIndex) {
+          newRow = {...row, isOpen: isOpen};
+        } else if (index === (rowIndex + 1)) {
+          // set the ResultView on the child rows
+          newRow = {
+            ...row,
+            cells: [
+              {title:
+                  <ResultView
+                    defaultTab={defaultTab}
+                    hideTestHistory={false}
+                    hideSummary={hideSummary}
+                    hideTestObject={hideTestObject}
+                    testResult={rows[rowIndex].result}
+                  />}
+            ]
+          };
+        } else {
+          newRow = {...row};
+        }
+        return newRow;
       });
+      setRows(updatedRows);
+    } else {
+      // handle updating isOpen on the clicked row (closing)
+      setRows(rows.map((row, index) => {
```

Comment:
instead of going through each row, we can find row that has been opened and set isOpen to true and ignore other rows:
```
      setRows(prevRows => {
        const updatedRows = [...prevRows];
        if (updatedRows[rowIndex]) {
          updatedRows[rowIndex] = { ...updatedRows[rowIndex], isOpen };
        }
        return updatedRows;
      });
```